dart-parser/src/parser/comment.rs

```diff
@@ -0,0 +1,77 @@
+use std::str;
+
+use nom::{
+    branch::alt,
+    bytes::complete::{is_not, tag},
+    combinator::{cut, eof, not, recognize},
+    multi::fold_many0,
+    sequence::{pair, preceded, terminated, tuple},
+    IResult, Parser,
+};
+
+use crate::{dart::comment::Comment, parser::common::*};
+
+pub fn comment(s: &str) -> IResult<&str, Comment> {
+    alt((
+        comment_single_line.map(Comment::SingleLine),
+        comment_multi_line.map(Comment::MultiLine),
+    ))(s)
+}
+
+fn comment_single_line(s: &str) -> IResult<&str, &str> {
+    recognize(tuple((tag("//"), is_not("\r\n"), alt((br, eof)))))(s)
+}
+
+fn comment_multi_line(s: &str) -> IResult<&str, &str> {
+    recognize(preceded(
+        tag("/*"),
+        cut(terminated(
+            recognize(fold_many0(
+                alt((
+                    is_not("*/"),
+                    pair(tag("*"), not(tag("/"))).map(|(g, _)| g),
+                    pair(tag("/"), not(tag("*"))).map(|(g, _)| g),
```

Comment:
I'd suggest to rewrite the two lines above like this:

```rust
terminated(tag("*"), not(char('/'))),
terminated(tag("/"), not(char('*'))),
```

(see [char()](https://docs.rs/nom/7.1.3/nom/character/complete/fn.char.html))